### DISCUSSION TITLE
Foundry Fundamentals > Introduction to window.ethereum> index.js fund function

### DISCUSSION THREAD

**Discussion by bxpana** [Cyfrin/foundry-full-course-cu#2721] [Q&A]
Where does the `await contract.fund(2, "0x0000000000000000000000000000000000000000",` come into play when the frontend is calling the fund function?

The ABI in the `constants.js` does not match the ABI from the deployed fundMe contract, but for some reason it works to fund via the frontend. I tried removing the `2, 0x00...` and putting in nothing and a blank array but both times got errors when trying to fund via the frontend. 

The only thing I can think of is that it has something to do with how ethers v6 works since the video for this lesson does not show these other inputs. Tried ChatGPT and it's also confused on why there is no consistency for the ABI.

**Comment by EngrPips** [comment]
This is an interesting question. I looked into that script now, and I am equally confused. Let's wait to hear from @PatrickAlphaC  when they are available.

**Reply by PatrickAlphaC** [reply]
@bxpana I think I had an issue with the code. Could you check now? It should be correct now. I used the code as an example to show how to verify malicious transactions, and use the UI section to do so. (You'll see it later in the course) and I left the codebase in a weird state. It should be good now.

**Reply by EngrPips** [reply]
Oh I see.

**Comment by rrocode** [comment]
Hello, need some clarifications too. Maybe i've missed that in tons of information in between :)

So, i understand what ABI is, but why do we need to pass it to metamask in e.g. here? 
```js
const contract = new ethers.Contract(contractAddress, abi, signer)
``` 

is it to make .js to understand the .fund() function below?
```js
 try {
      const transactionResponse = await contract.fund({
        value: ethers.parseEther(ethAmount),
      })
      await transactionResponse.wait(1)
    } catch (error) {
      console.log(error)
    }
```

what am i missing? How does this work under the hood?

Thanks in advance for your input.

**Reply by EngrPips** [reply]
This line of code `const contract = new ethers.Contract(contractAddress, abi, signer)` is basically creating a contract object, and to form a contract object, you need both the `abi` and the `address` of the contract. The address will give the details about where the function calls will be sent, while the `abi` will give information on how to encode the function call that will be sent to the address.

**Reply by rrocode** [reply]
Aha, ok, understand. So, ethers.Contract(...) is an abstract method that can accept any abi+address as parameters so that later on to be able to invoke their methods on created instance. Thanks!

**Reply by EngrPips** [reply]
Yeah, you get the gist.
